src/gui/qmlutils.cpp

```diff
@@ -14,16 +14,22 @@
 
 #include "gui/qmlutils.h"
 
+#include "common/asserts.h"
 #include "resources/resources.h"
 
 #include <QMessageBox>
 #include <QQmlContext>
 #include <QQuickItem>
 #include <QQuickWidget>
 
-void OCC::QmlUtils::initQuickWidget(OCQuickWidget *widget, const QUrl &src, QObject *ocContext)
+void OCC::QmlUtils::initQuickWidget(OCQuickWidget *widget, const QUrl &src, QObject *ocContext, QWidget *parent)
 {
+    if (!parent) {
+        parent = qobject_cast<QWidget *>(ocContext);
+        Q_ASSERT_X(parent, Q_FUNC_INFO, "If invoked without an explicit parent widget, ocContext is required to be a QWidget");
+    }
```

Comment:
If the context object is not a widget, we could transfer the ownership to qml.
What do you think? How would we then document that....